src/panels/lovelace/cards/hui-alarm-panel-card.ts

```diff
@@ -0,0 +1,313 @@
+    if (changedProps.has("_config")) {
+      return true;
+    }
+
+    const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
+    if (oldHass) {
+      return (
+        oldHass.states[this._config!.entity] !==
+        this.hass!.states[this._config!.entity]
+      );
+    }
+    return true;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+    const stateObj = this.hass.states[this._config.entity];
+
+    return html`
+      ${this.renderStyle()}
+      <ha-card
+        .header="${this._config.name || this._label(stateObj.state)}"
+        class="${classMap({ "not-found": !stateObj })}"
+      >
+        ${
+          !stateObj
+            ? html`
+                <div class="not-found">
+                  Entity not available: ${this._config.entity}
+                </div>
+              `
+            : html`
+                <ha-label-badge
+                  class="${classMap({ [stateObj.state]: true })}"
+                  .icon="${ICONS[stateObj.state] || "hass:shield-outline"}"
+                  .label="${this._stateIconLabel(stateObj.state)}"
+                ></ha-label-badge>
+                ${
+                  stateObj.state === "disarmed" && this._config!.states
+                    ? html`
+                        <div id="armActions" class="actions">
+                          ${
+                            this._config.states!.map((state) => {
+                              return html`
+                                <paper-button
+                                  noink
+                                  raised
+                                  .action="${state}"
+                                  @click="${this._handleActionClick}"
+                                  >${this._label(state)}</paper-button
+                                >
+                              `;
+                            })
+                          }
+                        </div>
+                      `
+                    : html`
+                        <div id="disarmActions" class="actions">
+                          <paper-button
+                            noink
+                            raised
+                            .action="${"disarm"}"
+                            @click="${this._handleActionClick}"
+                            >${this._label("disarm")}</paper-button
+                          >
+                        </div>
+                      `
+                }
+                <paper-input
+                  label="Alarm Code"
+                  type="password"
+                  .value="${this._code}"
+                ></paper-input>
+                <div id="keypad">
+                  ${
+                    BUTTONS.map((value) => {
+                      return !value
+                        ? html`
+                            <paper-button disabled></paper-button>
```

Comment:
Do we really just render a disabled button and not just a `<span></span>` ?

Reply:
Makes sure the spacing always the same so it doesnt render poorly. Seeing if I can get the span to take the shape of a button

Reply:
Making it a span require a lot of CSS to make it dynamic while the button already has it buiilt in. unless I am missing something. Applying my same CSS from the button doesnt work